src/logging.rs

```diff
@@ -0,0 +1,63 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use log::{Record, Metadata};
+use std::ffi::CString;
+use std::os::raw::c_char;
+use std::os::raw::c_int;
+
+pub type LogCallback = unsafe extern "C" fn(msg: *const c_char);
+
+struct LogProxy {level: log::Level, callback: LogCallback}
+
+impl log::Log for LogProxy {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= self.level
+    }
+
+    fn log(&self, record: &Record) {
+        if self.enabled(record.metadata()) {
+            let msg = format!("{} - {}", record.level(), record.args());

```

Comment:
I made the format such that the receiving end can decide what to do. But now that I read this back, we could instead give the callback two args `(level, msg)`.

Reply:
yes, please, let's return 2 arguments back